third_party/move/move-compiler-v2/src/env_pipeline/lambda_lifter.rs

```diff
@@ -247,21 +297,39 @@ impl<'a> ExpRewriterFunctions for LambdaLifter<'a> {
         // parameter indices in the lambda context to indices in the lifted
         // functions (courtesy of #12317)
         let mut param_index_mapping = BTreeMap::new();
-        for (used_param_count, (param, node_id)) in
+        for (used_param_count, (param, var_info)) in
             mem::take(&mut self.free_params).into_iter().enumerate()
         {
             let name = self.fun_env.get_local_name(param);
-            let ty = env.get_node_type(node_id);
-            let loc = env.get_node_loc(node_id);
+            let ty = env.get_node_type(var_info.node_id);
+            let loc = env.get_node_loc(var_info.node_id);
+            if var_info.modified {
+                env.error(
+                    &loc,
+                    &format!(
+                        "captured parameter `{}` cannot be modified inside of a lambda",
```

Comment:
You should probably also call this a "captured variable" here, since parameters can appear in an `Assign` expression as a symbol and will be treated as a local.  Determining whether the `Assign` symbol is a parameter or not requires tracking var scoping outside of the lambda, so probably isn't worth it.  Easier to just change the error message to be consistent.

Reply:
Done